igvm/commands.py

```diff
@@ -385,103 +389,105 @@ def host_info(vm_hostname):
             )
-            info[result_key] = red(simple_stats)
-            return
 
-        assert free >= 0 and free <= capacity
-        ratio = 1 - float(free) / float(capacity)
-        if ratio >= 0.9:
-            color = red
-        elif ratio >= 0.8:
-            color = yellow
-        else:
-            color = green
-
-        max_bars = 20
-        num_bars = int(round(ratio * max_bars))
-        info[result_key] = (
-            '[{}{}] {}%\n{}'
-            .format(
-                color('#' * num_bars), ' ' * (max_bars - num_bars),
-                int(round(ratio * 100)),
-                simple_stats,
+            if not 0 <= free <= capacity > 0:
+                log.warning(
+                    '{} ({}) and {} ({}) have weird ratio, skipping progress '
+                    'calculation'
+                    .format(free_key, free, capacity_key, capacity)
+                )
+                info[result_key] = red(simple_stats)
+                return
+
+            assert free >= 0 and free <= capacity
+            ratio = 1 - float(free) / float(capacity)
+            if ratio >= 0.9:
+                color = red
+            elif ratio >= 0.8:
+                color = yellow
+            else:
+                color = green
+
+            max_bars = 20
+            num_bars = int(round(ratio * max_bars))
+            info[result_key] = (
+                '[{}{}] {}%\n{}'
+                .format(
+                    color('#' * num_bars), ' ' * (max_bars - num_bars),
+                    int(round(ratio * 100)),
+                    simple_stats,
+                )
             )
-        )
 
-    _progress_bar('memory_free', 'memory', 'memory', 'MiB')
-    _progress_bar('disk_free_gib', 'disk_size_gib', 'disk', 'GiB')
+        _progress_bar('memory_free', 'memory', 'memory', 'MiB')
+        _progress_bar('disk_free_gib', 'disk_size_gib', 'disk', 'GiB')
 
-    max_key_len = max(len(k) for k in info.keys())
-    for category, keys in categories:
-        # Handle 'Other' section by defaulting to all keys
-        keys = list(keys or info.keys())
+        max_key_len = max(len(k) for k in info.keys())
+        for category, keys in categories:
+            # Handle 'Other' section by defaulting to all keys
+            keys = list(keys or info.keys())
 
-        # Any info available for the category?
-        if not any(k in info for k in keys):
-            continue
-
-        print('')
-        print(white(category, bold=True))
-        for k in keys:
-            if k not in info:
+            # Any info available for the category?
+            if not any(k in info for k in keys):
                 continue
 
-            # Properly re-indent multiline values
-            value = str(info.pop(k))
-            value = ('\n' + ' ' * (max_key_len + 3)).join(value.splitlines())
-            print('{} : {}'.format(k.ljust(max_key_len), value))
+            print('')
+            print(white(category, bold=True))
```

Comment:
Shouldn't we better use the logger for such stuff.

Reply:
I did not really write this code, just indented it to form a block after `with`. I can fix it while I'm at it, though, if you wish.

Reply:
This command is kind special.  It prints information about the VM.  Those are not really logs.

Reply:
That is true, it is the "info" function. I would prefer to not touch it right now yet.